extensions/amp-a4a/0.1/test/test-legacy-signature-verifier.js

```diff
@@ -0,0 +1,212 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FetchMock, networkFailure} from './fetch-mock';
+import {
+  data as validCSSAmp,
+} from './testdata/valid_css_at_rules_amp.reserialized';
+import {LegacySignatureVerifier} from '../legacy-signature-verifier';
+
+describes.realWin('LegacySignatureVerifier', {amp: true}, env => {
+
+  let fetchMock;
+  let keysetBody;
+  let verifier;
+  let result;
+
+  beforeEach(() => {
+    fetchMock = new FetchMock(env.win);
+    fetchMock.getOnce(
+        'https://cdn.ampproject.org/amp-ad-verifying-keyset.json',
+        () => keysetBody, {name: 'keyset'});
+    keysetBody = `{"keys":[${validCSSAmp.publicKey}]}`;
+    verifier = new LegacySignatureVerifier(env.win);
+    result = verifier.keys_;
+  });
+
+  afterEach(() => {
+    fetchMock./*OK*/restore();
+  });
+
+  function verifyIsKeyInfo(keyInfo) {
+    expect(keyInfo).to.be.ok;
+    expect(keyInfo).to.have.all.keys(
+        ['signingServiceName', 'hash', 'cryptoKey']);
+    expect(keyInfo.signingServiceName).to.be.a('string').and.not.to.equal('');
+    expect(keyInfo.hash).to.be.instanceOf(Uint8Array);
+  }
+
+  it('should fetch a single key', () => {
+    expect(result).to.be.empty;
+    verifier.loadKeyset('google', Promise.resolve());
+    expect(result).to.be.instanceof(Array);
+    expect(result).to.have.lengthOf(1);
+    return Promise.all(result).then(serviceInfos => {
+      expect(fetchMock.called('keyset')).to.be.true;
+      const serviceInfo = serviceInfos[0];
+      expect(serviceInfo).to.have.all.keys(['signingServiceName', 'keys']);
+      expect(serviceInfo['signingServiceName']).to.equal('google');
+      expect(serviceInfo['keys']).to.be.an.instanceof(Array);
+      expect(serviceInfo['keys']).to.have.lengthOf(1);
+      const keyInfoPromise = serviceInfo['keys'][0];
+      expect(keyInfoPromise).to.be.an.instanceof(env.win.Promise);
+      return keyInfoPromise.then(keyInfo => {
+        verifyIsKeyInfo(keyInfo);
+      });
+    });
+  });
+
+  it('should wait for promise', () => {
+    expect(result).to.be.empty;
+    let resolveWaitFor;
+    verifier.loadKeyset('google', new Promise(resolve => {
+      resolveWaitFor = resolve;
+    }));
+    expect(fetchMock.called('keyset')).to.be.false;
+    resolveWaitFor();
+    return Promise.all(result).then(() => {
+      expect(fetchMock.called('keyset')).to.be.true;
+    });
+  });
+
+  it('should fetch multiple keys', () => {
+    // For our purposes, re-using the same key is fine.
+    keysetBody =
+        `{"keys":[${validCSSAmp.publicKey},${
+                                               validCSSAmp.publicKey
+                                             },${validCSSAmp.publicKey}]}`;
```

Comment:
The linter looks at test files, but it doesn't care what you do inside template literals. `clang-format` is terrible at this. I'll make it less ugly.